specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -207,17 +212,47 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
             }
           );
         }
+        // Flag as invalid if no determinations are checked as current
         if (
           determination.collection !== undefined &&
           !determination.collection.models.some(
             (c: SpecifyResource<Determination>) => c.get('isCurrent')
           )
         ) {
-          determination.set('isCurrent', true);
+          return {
+            isValid: false,
+            reason: resourcesText.currentDeterminationRequired(),
+            saveBlockerKey: CURRENT_DETERMINATION_KEY,
+            resource: determination.collection.related,
+          };
         }
-        return { isValid: true };
+        return {
+          isValid: true,
+          saveBlockerKey: CURRENT_DETERMINATION_KEY,
+          resource: determination.collection?.related,
+        };
       },
     },
+    onRemoved: (determination: SpecifyResource<Determination>): void => {
+      // Block save when a current determination is deleted
+      if (determination.get('isCurrent'))
+        setSaveBlockers(
+          determination.collection?.related ?? determination,
+          determination.specifyTable.field.isCurrent,
+          [resourcesText.currentDeterminationRequired()],
+          CURRENT_DETERMINATION_KEY
+        );
+    },
+    onAdded: (determination: SpecifyResource<Determination>): void => {
+      // Clear any existing save blocker on adding a new current determination
+      if (determination.get('isCurrent'))
+        setSaveBlockers(
+          determination.collection?.related ?? determination,
+          determination.specifyTable.field.isCurrent,
+          [],
+          CURRENT_DETERMINATION_KEY
+        );
+    },
```

Comment:
(optional) 
If a little refactoring can be done, the code from Determination `customInit` should really be simplified and included here, in the `onAdded` callback. 

https://github.com/specify/specify7/blob/4e5722aabb4ae3e8afd2ee487f4bea68eaff83d1/specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts#L149-L165

Reply:
This is very verbose indeed. Also this part is duplicated here and in isCurrent fieldCheck and could be written into a function
```
if (determinaton.collection !== undefined) { 
         determinaton.collection.models.forEach( 
           (other: SpecifyResource<Determination>) => { 
             if (other.cid !== determinaton.cid) 
               other.set('isCurrent', false); 
           } 
         ); 
       } 
```

Reply:
Looks like we don't need to uncheck other determinations in `customInit` at all. If we set `isCurrent` to true when creating a determination, it triggers the `isCurrent` fieldCheck which then unchecks isCurrent for other determinations in the collection